networking/gcp/vpc/main.tf

```
@@ -0,0 +1,100 @@
+locals {
+  pod_subnets                    = can(coalesce(var.gke_name)) ? var.pod_subnets : []
+  public_subnets                 = var.enable_external_access ? var.public_subnets : []
+}
+
+# VPC
+resource "google_compute_network" "vpc" {
+  name                            = var.name
+  project                         = var.project
+  auto_create_subnetworks         = false
+  delete_default_routes_on_create = true
+  routing_mode                    = "GLOBAL"
+}
+
+# Private subnets
+resource "google_compute_subnetwork" "private_subnets" {
+  count                            = length(var.private_subnets)
+
+  name                             = "${var.name}-private-subnet-${count.index}"
+  project                          = var.project
+  region                           = var.region
+  network                          = google_compute_network.vpc.name
+  ip_cidr_range                    = var.private_subnets[count.index]
+  private_ip_google_access = var.enable_google_access
```

```suggestion
  for_each                            = toset(var.private_subnets)
  name                             = "${var.name}-private-subnet-${index(var.private_subnets, each.key)}"
  project                          = var.project
  region                           = var.region
  network                          = google_compute_network.vpc.name
  ip_cidr_range                    = each.key
  private_ip_google_access = var.enable_google_access
```

---

I completely agree on the `for_each`, but we should remove entirely the notion of index in the name. Indeed, the name is immutable, and if we add a new subnet, it might change the index of other subnets.

```suggestion
  for_each                 = var.private_subnets
  name                     = "${var.name}-private-subnet-${replace(each.key, "/\W/", "_")}"
  project                  = var.project
  region                   = var.region
  network                  = google_compute_network.vpc.name
  ip_cidr_range            = each.key
  private_ip_google_access = var.enable_google_access
```